lib/dai-plugin-mcd/src/QueryApiMcdService.js

```diff
@@ -0,0 +1,142 @@
+import { PublicService } from '@makerdao/services-core';
+import fetch from 'isomorphic-fetch';
+import { ServiceRoles } from './constants';
+import { RAY } from './constants';
+import { MDAI } from './index';
+import BigNumber from 'bignumber.js';
+import { stringToBytes } from './utils';
+import { TESTNET_ID } from '../../../contracts/networks';
+
+const LOCAL_URL = 'http://localhost:5000/graphql';
+const KOVAN_SERVER_URL = 'http://vdb0-testing.vulcanize.io:5000/graphql';
+
+export default class QueryApi extends PublicService {
+  constructor(name = ServiceRoles.QUERY_API) {
+    super(name, ['web3']);
+  }
+
+  connect() {
+    const network = this.get('web3').network;
+    switch (network) {
+      case TESTNET_ID:
+        this.serverUrl = LOCAL_URL;
+        break;
+      case 'kovan':
+      case 42:
+      default:
+        this.serverUrl = KOVAN_SERVER_URL;
+        break;
+    }
+  }
+
+  async _blockNumberToDateTime(blockNumber) {
+    const timestamp = (await this.get('web3').getBlock(blockNumber)).timestamp;
+    return new Date(1000 * timestamp);
+  }
+
+  //TODO: add error handling, and handle no return value from vulcanize?
+  async getCdpEventsForIlkAndUrn(
+    ilk,
+    urn,
+    currency,
```

Comment:
Having `currency` as an argument here and in `getPriceHistoryForPip` bugs me, because the MCD plugin as a whole is smart enough to figure out what currency corresponds to an ilk. But that functionality is most suited to the CdpTypeService. I think the right thing to do might be: have QueryApi just handle GraphQL requests/responses, no output formatting, and then have methods in CdpTypeService that take the raw responses from the QueryApi and apply the correct formatting.